dev/maven-plugin/src/main/scala/com/lightbend/lagom/maven/ConfigureMojo.scala

```diff
@@ -0,0 +1,23 @@
+package com.lightbend.lagom.maven
+
+import javax.inject.Inject
+
+import org.apache.maven.execution.MavenSession
+import org.apache.maven.plugin.AbstractMojo
+
+import scala.beans.BeanProperty
+
+/**
+ * Internal goal, invoked by other Lagom mojos that work with multiple projects at once, to read plugin configuration
+ * for a project and set up the projects context values.
+ */
+class ConfigureMojo @Inject() (session: MavenSession) extends AbstractMojo {
+
+  @BeanProperty
+  var lagomService: Boolean = _
```

Comment:
It can't be, mojos aren't passed around. And, since it's not a singleton, it will be instantiated for each time it's used, and it's use is only ever in a single threaded manner.